### PR TITLE
config(provider/openstack): Added extra settings and some documentation

### PR DESCRIPTION
**Pull request by kkapelon**

### CODE REVIEW

**Comment by kkapelon** [comment]
ping @spinnaker/reviewers

File: config/default-spinnaker-local.yml [hunk 1]
```diff
@@ -143,7 +146,10 @@ providers:
       username: ${OS_USERNAME}
       password: ${OS_PASSWORD}
       projectName: ${OS_PROJECT_NAME}
+      domainName: ${OS_PROJECT_DOMAIN_NAME}
```
Comment (kkapelon) [inline config/default-spinnaker-local.yml]: I have both properties OS_USER_DOMAIN_NAME and OS_PROJECT_DOMAIN_NAME defined locally (to the same value). Maybe you are correct and one of them is not needed. They are both valid however. https://docs.openstack.org/developer/python-openstackclient/authentication.html. Maybe add both of them? (With one of them commented out?)

Comment (ewiseblatt) [inline config/default-spinnaker-local.yml]: while you may have both, it seems overspecified to be using both and seems wrong to have one file overwrite the other. Pick one as your interface, or create an attribute for each and do the right thing internally to support that. I'm not familiar with open stack to tell you what the right thing is.

Keep in mind that spinnaker.yml will always be present. That is your base interface.
The default-spinnaker-local.yml is purely optional and is only used to default an initial spinnaker-local.yml if one was not explicitly supplied. So you cannot count on anything in it showing up in the runtime.

**Comment by kkapelon** [comment]
Updated setting.

**Pull request merged**